phoenix-core/src/it/java/org/apache/phoenix/end2end/DisableSplitForTaskTableIT.java

```diff
@@ -0,0 +1,134 @@
+
+        public PhoenixUpgradeCountingServices(
+                QueryServices services,
+                PhoenixEmbeddedDriver.ConnectionInfo connectionInfo,
+                Properties info) {
+            super(services, connectionInfo, info);
+        }
+
+        @Override
+        protected void setUpgradeRequired() {
+            super.setUpgradeRequired();
+        }
+
+        @Override
+        protected long getSystemTableVersion() {
+            return systemTableVersion;
+        }
+
+        @Override
+        protected boolean isInitialized() {
+            return !reinitialize && super.isInitialized();
+        }
+    }
+
+    public static class PhoenixUpgradeCountingDriver
+            extends PhoenixTestDriver {
+        private ConnectionQueryServices connectionQueryServices;
+        private final ReadOnlyProps overrideProps;
+
+        public PhoenixUpgradeCountingDriver(ReadOnlyProps props) {
+            overrideProps = props;
+        }
+
+        @Override
+        public boolean acceptsURL(String url) {
+            return true;
+        }
+
+        @Override
+        public synchronized ConnectionQueryServices getConnectionQueryServices(
+                String url, Properties info) throws SQLException {
+            if (connectionQueryServices == null) {
+                connectionQueryServices =
+                    new PhoenixUpgradeCountingServices(
+                        new QueryServicesTestImpl(getDefaultProps(),
+                            overrideProps), ConnectionInfo.create(url), info);
+                connectionQueryServices.init(url, info);
+            } else if (reinitialize) {
+                connectionQueryServices.init(url, info);
+                reinitialize = false;
+            }
+            return connectionQueryServices;
+        }
+    }
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> props = Maps.newConcurrentMap();
+        props.put(BaseTest.DRIVER_CLASS_NAME_ATTRIB,
+            PhoenixUpgradeCountingDriver.class.getName());
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+    }
+
+    @Test
+    public void testUpgradeOnlyHappensOnce() throws Exception {
+        ConnectionQueryServices services = DriverManager.getConnection(getUrl())
+            .unwrap(PhoenixConnection.class).getQueryServices();
+        assertTrue(services instanceof PhoenixUpgradeCountingServices);
+        reinitialize = true;
+        systemTableVersion = MetaDataProtocol.MIN_SYSTEM_TABLE_TIMESTAMP;
+        DriverManager.getConnection(getUrl());
+        try (Admin admin = services.getAdmin()) {
+            String taskSplitPolicy = admin
+                .getDescriptor(TableName.valueOf(
+                    PhoenixDatabaseMetaData.SYSTEM_TASK_NAME))
+                .getRegionSplitPolicyClassName();
+            assertEquals(DisabledRegionSplitPolicy.class.getName(),
+                taskSplitPolicy);
+        }
```

Comment:
Do we really need a separate IT for this? There is an existing IT that uses the upgrade counting driver and services. Let's reuse that instead